nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/work/manager/WorkManagerObject.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management.work.manager;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.core.operations.services.workmanager.WorkManagerRunWorkInFailure;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.management.api.ProbeInfo;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 11.1
+ */
+@WebObject(type = "workmanager")
+public class WorkManagerObject extends DefaultObject {
+
+    public static final String TIMEOUT_SECONDS_PARAM_KEY = "timeoutSeconds";
+
+    /**
+     * Re-executes works in this DLQ for a repair purpose.
+     *
+     * @param probeName the shortcut name of the probe to launch
+     * @return the result of the probe in a {@link ProbeInfo}
+     * @throws OperationException
+     * @throws IOException
+     */
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("runworksinfailure")
+    public String launch(@FormParam(TIMEOUT_SECONDS_PARAM_KEY) String timeoutSeconds)
+            throws OperationException, IOException {
+        AutomationService automationService = Framework.getService(AutomationService.class);
+        try (OperationContext operationalCtx = new OperationContext(ctx.getCoreSession())) {
+            if (timeoutSeconds != null) {
+                Map<String, Serializable> params = new HashMap<>();
+                params.put(TIMEOUT_SECONDS_PARAM_KEY, Long.parseLong(timeoutSeconds));
+                return ((Blob) automationService.run(operationalCtx, WorkManagerRunWorkInFailure.ID,
+                        params)).getString();
+            } else {
+                // To avoid logging useless warns for every call with no param
```

Comment:
Why not just passing an empty `Map` if there is no `timeoutSeconds` param set, so that we can keep the same call?

Reply:
Because you get a warn logged while you can avoid it by passing nothing.
This is covered in the test (the second half)

Reply:
code will be replaced by : 
```
try (OperationContext operationalCtx = new OperationContext(ctx.getCoreSession())) {
            Map<String, Serializable> params = new HashMap<>();
            if (timeoutSeconds != null) {
                params.put(TIMEOUT_SECONDS_PARAM_KEY, timeoutSeconds);
            }

            return ((Blob) automationService.run(operationalCtx, WorkManagerRunWorkInFailure.ID,
                    params)).getString();

        }
```